### DISCUSSION TITLE
How to use Vite plugin?

### DISCUSSION THREAD

**Discussion by dungonfq** [nuxt/nuxt#17566] [Questions] [answered]
Hi team,

I'm trying to integrate Sentry w Nuxt3. Since Nuxt3 hasn't supported [the Sentry module](https://sentry.nuxtjs.org/) yet, I need to config it manually. To push the source map to Sentry, I found a Vite plugin https://github.com/ikenfin/vite-plugin-sentry.

How can I add a Vite plugin into Nuxt3? Because, seem like [nuxt.config.vite](https://v3.nuxtjs.org/api/configuration/nuxt.config#vite) doesn't support `plugin` option.

In short, `vite-plugin-sentry` will generated `.map` files and match associated .js files in build times.

Thank you a lot.

**Comment by qnp** [comment, marked as answer]
You need to create a nuxt module. I am currently doing experiments and will come soon with a solution.
Based on my current experiments, I created a nuxt module and used the `@nuxt/kit` method `addVitePlugin` to add the `vite-plugin-sentry`, in combination with `extendViteConfig` to set `config.build.sourcemap = true`.

Yet, this solution is not working since the source map files seems to be generated somewhere in memory or in a tmp folder, until the end of the nuxt build process where they are copied to `.output/public/_nuxt`, so the plugin does not find any file on time.

So I'm trying to make the calls to Sentry (as `vite-plugin-sentry`) but in a late nuxt hook (_e.g._ 'close' or 'build:done')

Keep in touch

**Reply by dungonfq** [reply]
Thank @qnp very much, I will try it. I also came up with idea using `addVitePlugin` recently

**Comment by qnp** [comment]
Synthetizing [vite-plugin-sentry](https://github.com/ikenfin/vite-plugin-sentry) for the build steps and based on [this answer](https://github.com/nuxt-community/sentry-module/issues/358#issuecomment-1016983543) for the runtime plugin, I came up with this nuxt module. We will test it for a while in our company before considering releasing it as a public npm package for the community.

=> [sentry.zip](https://github.com/nuxt/framework/files/8719364/sentry.zip)

Don't hesitate if you have questions and feedbacks on it.

**Reply by dungonfq** [reply]
Hi @qnp, I tried ur modules. Everything is ok except all my `.mjs` file was not uploaded to Sentry. Nuxt built into `.mjs`, does it work well on ur side?

![artifact](https://user-images.githubusercontent.com/57652048/169451458-d8e6b2a3-b90d-4975-b59a-c6e58bed494f.png)

---
Updated:
I found a option in `SentryCliUploadSourceMapsOptions`:
```js
/**
 * This sets the file extensions to be considered.
 * By default the following file extensions are processed: js, map, jsbundle and bundle.
 */
ext?: string[];
```

So my config for Nuxt3 should be:
```js
const sourceMaps: SentryCliUploadSourceMapsOptions = {
      include: [path.join(buildDir, clientDir)],
      ignore: ['node_modules'],
      urlPrefix: `~${clientDir}`,
      ext: ['js', 'mjs', 'map'],
}
```

**Reply by dungonfq** [reply]
@qnp Another problem, Sentry serves the source maps from `~/_nuxt`, not `~/public/_nuxt`. So at first, the minified file could not find its `.map`
![minified-file-could-not-found-map](https://user-images.githubusercontent.com/57652048/169454739-0d6bac73-63bb-4f8b-9181-3fb9604a2268.png)

Therefore I changed my `urlPrefix: '~/_nuxt'` and it worked well.
![minified-file-found-its-map](https://user-images.githubusercontent.com/57652048/169454776-6304837d-002f-40f2-9799-8e72ac921f08.png)

My source maps are now correct:
![artifact_correct](https://user-images.githubusercontent.com/57652048/169454885-cd9c5fbe-0eea-4600-b6c0-fde19e810af3.png)

Thank u for ur module. It saved my day 🌟

**Comment by qnp** [comment]
Hi @dungonfq , thanks for your feedback 😉 
I will update that on my side 👍
